tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -0,0 +1,155 @@
+import { useBackend } from '../backend';
+import { Button, DepartmentEntry, DepartmentPane, Icon, Stack } from '../components';
+import { Window } from '../layouts';
+import { Color } from 'common/color';
+import { SFC } from 'inferno';
+import { JobToIcon } from './common/JobToIcon';
+
+type BaseVars = {
+  name: string;
+  open_slots: number;
+};
+
+type Job = BaseVars & {
+  unavailable_reason: string | null;
+  command: boolean;
+  used_slots: number;
+  icon: string;
+  prioritized: boolean;
+  job_description: string;
+};
+
+type Department = BaseVars & {
+  jobs: Job[];
+  open_slots: number;
+  color: string;
+};
+
+type Data = {
+  departments: Department[];
+  alert_state: string;
+  shuttle_status: string;
+  disable_jobs_for_non_observers: boolean;
+  priority: boolean;
+  round_duration: string;
+};
+
+// Specifically not typed for flexibility.
+export const JobEntry: SFC<{ job; department; act: Function }> = (data) => {
+  const job = data.job;
+  const department = data.department;
+  return (
+    <Stack.Item fill>
+      <Button
+        width="100%"
+        style={{
+          // Try not to think too hard about this one.
+          'background-color': job.unavailable_reason
+            ? 'lightgrey'
+            : job.prioritized
+              ? '#308d25'
```

Comment:
This should be a color from the colors file, or at least comment what this is supposed to be

Reply:
The colours file has too limited a palette for me to pick out a green that won't hurt folk's eyes when compared to department colours, so I'll comment on what the colour should be. I'm a little surprised these don't have the colour box next to them like DM does.

Reply:
Unless you want me to move the colours into the colour constants so they're properly named.

Reply:
Still need concrete feedback on this one.